op-acceptance-tests/tests/base/chain_test.go

```diff
@@ -1,48 +1,67 @@
 package base
 
 import (
+	"context"
+	"fmt"
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/require"
-
 	"github.com/ethereum-optimism/optimism/devnet-sdk/system"
 	"github.com/ethereum-optimism/optimism/devnet-sdk/testing/systest"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/wait"
 	"github.com/ethereum-optimism/optimism/op-service/testlog"
 	"github.com/ethereum/go-ethereum/log"
 )
 
 // TestChainFork checks that the chain does not fork.
 func TestChainFork(t *testing.T) {
-	chainIdx := uint64(0)
-
 	systest.SystemTest(t,
-		chainForkTestScenario(chainIdx),
+		chainForkTestScenario(),
 	)
 }
 
-func chainForkTestScenario(chainIdx uint64) systest.SystemTestFunc {
+func chainForkTestScenario() systest.SystemTestFunc {
 	return func(t systest.T, sys system.System) {
-		const numCyclesToWait = 5
-
 		logger := testlog.Logger(t, log.LevelInfo)
 		logger.Info("Started test")
 
-		chain := sys.L2s()[chainIdx]
+		// Check all L2 chains
+		for i, chain := range sys.L2s() {
+			chainIndex := i
+			currentChain := chain
+			t.Run(fmt.Sprintf("Chain_%d", chainIndex), func(t systest.T) {
+				t.Parallel()
+				chainLogger := logger.New("chain", chainIndex)
+
+				// Initial chain fork check
+				laterCheck, err := systest.CheckForChainFork(t.Context(), currentChain, chainLogger)
+				if err != nil {
+					t.Fatalf("first chain fork check failed: %v", err)
+				}
 
-		// Check for a chain fork
-		logger.Info("Checking for chain fork")
-		laterCheck, err := systest.CheckForChainFork(t.Context(), chain, logger)
-		require.NoError(t, err, "first chain fork check failed")
+				// Get a geth client
+				chainLogger.Info("Waiting for next block")
+				node := currentChain.Nodes()[0]
+				client, err := node.GethClient()
+				if err != nil {
+					t.Fatalf("failed to get geth client: %v", err)
+				}
 
-		// Wait for a potential chain fork
-		logger.Info("Waiting for a potential chain fork")
-		time.Sleep(numCyclesToWait * 2 * time.Second)
+				// Create a context with timeout
+				ctx, cancel := context.WithTimeout(t.Context(), 60*time.Second)
+				defer cancel()
 
-		// Check for a chain fork again
-		err = laterCheck(false)
-		require.NoError(t, err, "second chain fork check failed")
-		t.Log("Chain fork check passed")
+				// Get the current block number
+				currentBlockNum, err := client.BlockNumber(ctx)
+				if err != nil {
+					t.Fatalf("failed to get current block number: %v", err)
+				}
 
+				// Wait for the next block
+				targetBlockNum := currentBlockNum + 1
+				chainLogger.Debug("Waiting for block", "target", targetBlockNum)
+				wait.ForBlock(ctx, client, targetBlockNum)
```

Comment:
Sorry I missed the before - you can simplify this even further with `wait.ForNextBlock`

The wait package is full of really good stuff. :)

Reply:
And doesn't it need to call `laterCheck` after this wait?

Reply:
Yep! Just noticed the same issue; over-deleted code in my excitement to simplify it. :)
Re-added it. Thanks

Reply:
Oh, yes, neat. I'll switch to `wait.ForNextBlock`

Reply:
Will need to be `require.NoError(t, wait.ForNextBlock(...))`